src/common.c

```diff
@@ -2200,8 +2204,9 @@ static int map_file(mapped_file_t *mf, const char *path) {
 
   if (sizeof(st.st_size) != sizeof(size_t)) {
     // on 32 bit systems, check if there is an overflow
-    if (st.st_size > (off_t)SSIZE_MAX) {
-      fprintf(stderr, "mmap() size_t overflow for file %s\n", path);
+	if (st.st_size > (off_t)MAX_FILE_SIZE  /*1 GB*/ ) {
```

Comment:
https://github.com/stlink-org/stlink/issues/1214#issuecomment-1013743705

Reply:
It would be nice if we could include the related necessary changes.

Reply:
I’m not sure how to mark it up with GitHub suggestion syntax. I suggest:

- removing the `MAX_FILE_SIZE` `ifdef`/`define`/`endif`,
- replacing the `st.st_size > (off_t)SSIZE_MAX` test with `st.st_size > (intmax_t) SIZE_MAX`,
- perhaps removing the `sizeof(st.st_size) != sizeof(size_t)` test as well.

Reply:
Addressed in commit f6cfd1bfe3eb50b90befe03293808727c66cb8bd.